lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -151,8 +152,57 @@ def which(cmd):
     return None
 
 
-class MPI(object):
+class RunFileLoopMPI(RunFile, RunLoop):
+    """
+    Combination of RunFile and RunLoop to support output to file,
+    while also checking whether any output has been produced after a specified amount of time.
+    """
+    def __init__(self, cmd, **kwargs):
+        """
+        handle initialisation: get filename and output timeout from arguments
+        """
+        self.filename = kwargs.get('filename', None)
+        self.output_timeout = kwargs.pop('output_timeout', None)
+
+        # check timeout
+        try:
+            self.output_timeout = int(self.output_timeout)
+        except ValueError as err:
+            raise ValueError("Invalid timeout value: %s (should be integer): %s" % (self.output_timeout, err))
+
+        super(RunFileLoopMPI, self).__init__(cmd, **kwargs)
+
+        self.timeout_cnt = 1
+        self.seen_output = self.output_timeout<0 #no check when output_timeout is negative
 
+    def _loop_process_output(self, output):
+        """
+        check if process is generating any output at all; if not, warn the user after a set amount of time
+        """
+        if output:
+            raise ValueError("Output was found using RunFile:\n%s\n This means something went horribly wrong." % output)
+
+        if self.seen_output:
+            return
+        else:
+            with open(self.filename, 'r') as fin:
+                if fin.read():
+                    self.seen_output = True
+
+            time_passed = self._loop_count * self.LOOP_TIMEOUT_MAIN
+            if time_passed > self.output_timeout * self.timeout_cnt and not self.seen_output:
+                warning_msg = '\n'.join([
+                "WARNING: mympirun has been running for %s seconds without seeing any output." % time_passed,
+                "This may mean that your program is hanging, please check and make sure that is not the case!",
+                '',
+                "If this warning is printed too soon and the program is doing useful work without producing any output,",
+                "you can increase the timeout threshold via --output-check-timeout (current setting: %s seconds)" % self.output_timeout,
```

Comment:
line too long (over 120 chars)?

drop the `else` above, move indent to the left, and you're good I think :)